### PR TITLE
Solve compilation issues in GitHub Actions

### PR DESCRIPTION
**Pull request by allanleal**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,6 +19,9 @@ include(CondaAware)
 # Include the cmake variables with values for installation directories
 include(GNUInstallDirs)
 
+# Checking CMAKE_INSTALL_PREFIX
+message(STATUS "CMAKE_INSTALL_PREFIX: ${CMAKE_INSTALL_PREFIX}")
+
```
Comment (volpatto) [inline CMakeLists.txt:24]: This part should be removed. I put it here just to check things.

File: CMakeLists.txt [hunk 2]
```diff
@@ -59,6 +62,8 @@ add_subdirectory(Optima)
 # Build the python wrappers
 if(OPTIMA_BUILD_PYTHON)
     find_package(pybind11 REQUIRED)
+    find_package (Python COMPONENTS Interpreter Development)
+    message(STATUS "Python library directory: ${Python_SITEARCH}")
```
Comment (volpatto) [inline CMakeLists.txt:66]: It should be removed as well...

File: python/package/CMakeLists.txt [hunk 3]
```diff
@@ -6,27 +9,35 @@ set(OPTIMA_PYTHON_PACKAGE_PATH ${CMAKE_CURRENT_BINARY_DIR}/optima)
...
-set(SETUP_PY ${OPTIMA_PYTHON_PACKAGE_PATH}/setup.py) # ensure setup.py is inside the package dir
+set(SETUP_PY ${CMAKE_CURRENT_BINARY_DIR}/setup.py) # ensure setup.py is inside the package dir
 configure_file(${SETUP_PY_IN} ${SETUP_PY})
 
 # Create a custom target to build the python package during build stage
+#add_custom_target(optima-setuptools ALL
+#    COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_CURRENT_SOURCE_DIR}/optima ${OPTIMA_PYTHON_PACKAGE_PATH}
+#    COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:optima4py> ${OPTIMA_PYTHON_PACKAGE_PATH}
+#    COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY} --quiet install --force --prefix=${CMAKE_BINARY_DIR}
+#    WORKING_DIRECTORY ${OPTIMA_PYTHON_PACKAGE_PATH})
```
Comment (volpatto) [inline python/package/CMakeLists.txt:20]: I have commented here to try the same code as for Reaktoro. Actually, the problem is that the `optima` installation should be in `${CONDA_PREFIX}\Lib\site-packages`, but it has been installed in `${CONDA_PREFIX}\Library\lib\site-packages` or elsewhere.

File: python/package/CMakeLists.txt [hunk 4]
```diff
@@ -6,27 +9,35 @@ set(OPTIMA_PYTHON_PACKAGE_PATH ${CMAKE_CURRENT_BINARY_DIR}/optima)
...
 # Set the path where the python package is installed to CMAKE_INSTALL_PREFIX if not given
 if(NOT DEFINED OPTIMA_PYTHON_INSTALL_PREFIX)
     set(OPTIMA_PYTHON_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX})
+    # set(OPTIMA_PYTHON_INSTALL_PREFIX ${CONDA_PREFIX})  # installation should be in ${CONDA_PREFIX}/Lib/site-packages
 endif()
 
 # If the path is already in Windows format (with backslashes), it can't be
 # added directly to the string below. CMake complains about "Invalid escape sequence".
 file(TO_CMAKE_PATH "${OPTIMA_PYTHON_INSTALL_PREFIX}" OPTIMA_PYTHON_INSTALL_PREFIX)
+file(TO_CMAKE_PATH "${PYTHON_EXECUTABLE}" PYTHON_EXECUTABLE)
```
Comment (volpatto) [inline python/package/CMakeLists.txt:40]: This avoids "Invalid escape sequence" warnings as well.

File: python/package/setup.py.in [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
-setup(name='Optima',
+setup(name='optima',
       version='${PROJECT_VERSION}',
       description='The Python interface of the Optima library.',
       author='Allan Leal',
       author_email='allan.leal@erdw.ethz.ch',
       url='github.com/allanleal/optima',
       license='GNU LGPL v2.1',
       package_dir={'optima': ''},
-      packages=['optima', 'optima._extensions'],
```
Comment (volpatto) [inline python/package/setup.py.in:11]: So far we do not have extensions for `optima`, so I have removed related unused codes.

**Comment by volpatto** [comment]
Well, I got an even worse result with the last commits (sorry). To recover to a point where Linux and OSX are passing, go back to commit "Fix win cmake python install" (d1195b5)

**Comment by volpatto** [comment]
@allanleal a test is falling on Linux... no idea why.

**Comment by allanleal** [comment]
I fix the test error and restored the things you mentioned above plus a few more. I think we only need to get paths right in Windows now.

**Comment by allanleal** [comment]
Closing this PR, which was addressed in PR #7.

**Pull request closed without merging**